src/test/java/spoon/test/signature/SignatureTest.java

```diff
@@ -271,24 +275,42 @@ public void testUnboundFieldSignature(){
 		catch(Exception e){
 			//must fail
 		}
-		
+
 		CtClass<?> clazz1 = (CtClass<?>) factory.Type().getAll().get(0);
 		assertNotNull(clazz1);
-		
+
 		//**FIRST PART: passing local variable access.
 		///--------From the first method we take the method invocations
 		CtMethod<?> methodString = (CtMethod<?>) clazz1.getAllMethods().toArray()[0];
 
 		CtAssignment<?,?> invoToInt1 = (CtAssignment<?,?>) methodString.getBody().getStatement(0);
-	
+
 		String sigAssign = invoToInt1.getSignature();
-		
+
 		CtExpression<?> left = invoToInt1.getAssigned();
 		String sigleft = left.getSignature();
 		assertNotEquals("",sigleft.trim());
-	
+
 		assertTrue(sigAssign.contains("<no type>"));
-	
+
+	}
+
+	@Test
+	public void testArgumentNotNullForExecutableReference() throws Exception {
+		final Launcher launcher = new Launcher();
+		launcher.addInputResource("./src/test/resources/variable/PropPanelUseCase_1.40.java");
+		launcher.setSourceOutputDirectory("./target/trash");
+		launcher.getEnvironment().setNoClasspath(true);
+		launcher.run();
+
+		final List<CtExecutableReference> references = Query.getReferences(launcher.getFactory(), new ReferenceTypeFilter<CtExecutableReference>(CtExecutableReference.class) {
+			@Override
+			public boolean matches(CtExecutableReference reference) {
+				return "addField".equals(reference.getSimpleName()) && super.matches(reference);
+			}
+		});
+		for (CtExecutableReference reference : references) {
+			assertNotEquals("#addField(null, null)", reference.getSignature());
```

Comment:
Sometimes, there are types found by JDT. So, we can have `#addField(null, type)`, `#addField(type, null)` or `#addField(null, null)`.